### PR TITLE
[Transform] report last search time in transform stats

### LINKED ISSUE #66410: [CI] TransformContinuousIT.testContinousEvents failing
This started failing often recently. Also average runtime have increased in the same time period so something is going on here.

**Build scan**:
https://gradle-enterprise.elastic.co/s/f42ebecbqgebo/tests/:x-pack:plugin:transform:qa:multi-node-tests:javaRestTest/org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT/testContinousEvents

**Repro line**:
`./gradlew ':x-pack:plugin:transform:qa:multi-node-tests:javaRestTest' --tests "org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents" -Dtests.seed=CE925CC7CF87CC0F -Dtests.security.manager=true -Dtests.locale=ar-SD -Dtests.timezone=America/Cancun -Druntime.java=8`

**Reproduces locally?**:
Didn't reproduce for me.

**Applicable branches**:
`master` and `7.x`

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT&tests.disabledDistributions=WyJvdXRjb21lOnBhc3NlZCJd&tests.sortField=FAILED&tests.test=testContinousEvents&tests.unstableOnly=true

**Failure excerpt**:
```

org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT > testContinousEvents FAILED
    java.lang.AssertionError: transform [continuous-histogram-pivot-test] does not progress, state: INDEXING, reason: null
    Expected: a value greater than <1608062383906L>
         but: <1608062360729L> was less than <1608062383906L>
        at __randomizedtesting.SeedInfo.seed([CE925CC7CF87CC0F:F5A42DDC45EC4E59]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.lambda$waitUntilTransformsReachedUpperBound$3(TransformContinuousIT.java:499)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1014)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.waitUntilTransformsReachedUpperBound(TransformContinuousIT.java:497)
        at org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT.testContinousEvents(TransformContinuousIT.java:275)
```

### PR DESCRIPTION
**Pull request by hendrikmuhs**
transforms reports the the last time changes where detected with `changes_last_detected_at`, however that doesn't tell a user it searched for changes, this PR adds a field `last_search_time` to report when transform searched for changes the last time.

fixes #66410
relates #66367

### Notes
 - should be super useful for support cases
 - cat transforms should report `last_search_time` (#66367)
 - makes testing more reliable (#66410)
 - not reported for stopped transforms

### Review

the main changes are part of [`bd59629` (#66718)](https://github.com/elastic/elasticsearch/pull/66718/commits/bd596295acaec91ab4d66e0ab484b2a9cddbad0a), the rest are style and test fixes

### CODE REVIEW

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java [hunk 1]
```diff
@@ -355,6 +357,8 @@ protected void onStart(long now, ActionListener<Boolean> listener) {
             }));
         } else {
             hasSourceChanged = true;
+            context.setChangesLastSearchedAt(instantOfTrigger);
+            context.setChangesLastDetectedAt(instantOfTrigger);
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java:361]: This changes behaviour for `changed_last_detected_at`. The change seems correct but is it fixing another issue with that not getting updated properly?

Comment (hendrikmuhs) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java:361]: yes, the `if` is "continuous" mode, the `else` is "batch". I see no reason why we report  `changed_last_detected_at` (and `changed_last_searched_at`) only for continuous but not for batch. This makes it more consistent.

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointingInfo.java [hunk 2]
```diff
@@ -49,13 +49,13 @@
         a -> {
             long behind = a[2] == null ? 0L : (Long) a[2];
             Instant changesLastDetectedAt = (Instant) a[3];
-            Instant changesLastSearchedAt = (Instant) a[4];
+            Instant lastSearchTime = (Instant) a[4];
```
Comment (przemekwitek) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointingInfo.java:52]: `lastSearchTime` or `lastSearchedTime`?

Please use the same variable name in both `TransformCheckpointingInfo` classes.

File: x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java [hunk 3]
```diff
@@ -276,7 +275,7 @@ public void testContinousEvents() throws Exception {
 
             // at random we added between 0 and 999_999ns == (1ms - 1ns) to every data point, so we add 1ms, so every data point is before
             // the checkpoint
-            waitUntilTransformsReachedUpperBound(runDate.toEpochMilli() + 1, run);
+            waitUntilTransformsProcessedNewData(ContinuousTestCase.SYNC_DELAY, run);
```
Comment (przemekwitek) [inline x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java:278]: Just double-checking: Is "+1" no longer needed here?

**Review by przemekwitek — APPROVED** [review]
LGTM

File: x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java (hunk 3 above)
Comment (hendrikmuhs) [reply x-pack/plugin/transform/qa/multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/continuous/TransformContinuousIT.java:276]: I removed the outdated comment. The test waits according to `delay` with `greaterThan`, so it _should_ be fine.

**Pull request merged**